trucks.js

```diff
@@ -501,4 +501,196 @@ var foodTrucks = [
+    var todaysTrucks = _.filter(foodTrucks, function(item) {
+
+        // Exclude truck if object doesn't include 'schedule' prop, otherwise check for 'day' match
+        if (typeof item.schedule === 'undefined') {
+            return false;
+        } else {
+            return (item.schedule.indexOf(day) !== -1);
+        }        
+    });
+    
+    return todaysTrucks;
+}
+
+// Function returns all truck objects
+function getTrucks() {
+    return foodTrucks;
+}
+
+// Function returns object info for matching truck if found
+function getTruck(name) {
+    var matchingTruck = {};
+    
+    name = name.toLowerCase().trim();
+
+    // Use underscore to find matching truck
+    matchingTruck = _.find(foodTrucks, function (i) { 
+        if (i.name.toLowerCase().trim() === name) {
+            return i;
+        }
+    });
+    
+    // Return matchingTruck
+    return matchingTruck;
+}
+
+//Function returns array of unique food types
+function getFoodTypes() {
+    var uniqueFoods = [];
+    var index = 0;
+
+    // Returns list of unique food types for all trucks
+	uniqueFoods = _.uniq(_.flatten(_.pluck(foodTrucks, "type")));
+    
+    // Sorts list, case-insensitive (from stackoverflow)
+    uniqueFoods = _.sortBy(uniqueFoods, function (i) { return i.toLowerCase(); });
+
+    // Return array of unique food types
+    return uniqueFoods;
+  
+}
+
+// Function returns object info for matching food type (if found)
+function filterByFoodType(type) {
+    var matchingTrucks= [];
+    
+    type = type.toLowerCase();
+
+    /*
+     * Filter returns the trucks that sell a specific food type 'type'
+     * QUESTION:  I added a prototype method make it case insensitive -
+     * is there a better way?
+     */
+	var matchingTrucks = _.filter(foodTrucks, function (item) {
+                
+		if (typeof item.type === 'undefined') {
+			return false;
+		} else {
+            return (item.type.toLowerCase().indexOf(type) !== -1);
+		}
+	});
+
+    return matchingTrucks;    
+}
+
+
+// Prototype method returns a lowercase version of the array that calls it (found concept on Stackoverflow, changed somewhat)
+Array.prototype.toLowerCase = function() { 
+    var lcArray = [];
+    
+    for (var i = 0, l = this.length; i < l; i++) {
+        lcArray[i] = this[i].toLowerCase(); 
+    }
+    
+    return lcArray;
+}
+
+// Add a truck to the foodTrucks array
+function addTruck(truckObj) {
+    if (truckObj) {
+        foodTrucks.push(truckObj); 
+    }    
+}
+
+// Remove a truck from the foodTrucks array
+function removeTruck(name) {
+    var index = _.findIndex(foodTrucks, function (truck) {
+        return truck.name === name;
+    });
+    foodTrucks.splice(index, 1);
+}
+
+// Update info for a truck in the foodTrucks array
+function updateTruck(truckObj) {
+    if (truckObj) {    
+        var index = _.findIndex(foodTrucks, function (truck) {
+            return truck.name === truckObj.name;
+        });
+        foodTrucks[index] = truckObj;
```

Comment:
Way more efficient than the silly way I was bumbling about trying to do this in lecture.